lib/client.js

```diff
@@ -1901,6 +1902,71 @@ MatrixClient.prototype.scrollback = function(room, limit, callback) {
     return defer.promise;
 };
 
+/**
+ * Take an EventContext, and back/forward-fill results.
+ *
+ * @param {module:models/event-context.EventContext} eventContext  context
+ *    object to be updated
+ * @param {boolean} backwards  true to fill backwards, false to go forwards
+ *
+ * @return {module:client.Promise} Resolves: updated EventContext object
+ * @return {module:http-api.MatrixError} Rejects: with an error response.
+ */
+MatrixClient.prototype.paginateEventContext = function(eventContext, backwards) {
+    // TODO: we should implement a backoff (as per scrollback()) to deal more
+    // nicely with HTTP errors.
+
+    var token = eventContext.getPaginateToken(backwards);
+    if (!token) {
+        // no more results.
+        throw new Error("No paginate token");
```

Comment:
Undocumented `throw`. I generally would prefer if we *didn't* throw regardless, and instead return a rejected promise e.g. `return q.reject(someError)` because otherwise you're forcing clients to have to have 2 ways of handling error responses from this single call (rejected promises and a try/catch) which sucks.